backend/app/api/routes/responses.py

```diff
@@ -73,14 +86,165 @@ def get_file_search_results(response):
+        )
+
+        send_callback(request.callback_url, callback_response.model_dump())
+        logger.info(
+            f"[responses.process_response] Callback sent successfully, assistant={request.assistant_id}, project_id={request.project_id}, organization_id={organization_id}"
+        )
+
+
+@router.post("/responses", response_model=dict)
+async def responses(
+    request: ResponsesAPIRequest,
+    background_tasks: BackgroundTasks,
+    _session: Session = Depends(get_db),
+    _current_user: UserOrganization = Depends(get_current_user_org),
+):
+    """Asynchronous endpoint that processes requests in background."""
+    logger.info(
+        f"[responses.responses] Processing response request for assistant_id={request.assistant_id}, project_id={request.project_id}, organization_id={_current_user.organization_id}"
+    )
+
+    # Get assistant details
+    assistant = get_assistant_by_id(
+        _session, request.assistant_id, _current_user.organization_id
+    )
+    if not assistant:
+        logger.error(
+            f"[responses.responses] Assistant not found: assistant_id={request.assistant_id}, project_id={request.project_id}, organization_id={_current_user.organization_id}"
+        )
+        raise HTTPException(
+            status_code=404,
+            detail="Assistant not found or not active",
+        )
+
+    credentials = get_provider_credential(
+        session=_session,
+        org_id=_current_user.organization_id,
+        provider="openai",
+        project_id=request.project_id,
+    )
+    if not credentials or "api_key" not in credentials:
+        logger.error(
+            f"[responses.responses] OpenAI API key not configured for org_id={_current_user.organization_id}, project_id={request.project_id}, organization_id={_current_user.organization_id}"
+        )
+        return {
+            "success": False,
+            "error": "OpenAI API key not configured for this organization.",
+            "data": None,
+            "metadata": None,
+        }
+
+    client = OpenAI(api_key=credentials["api_key"])
+
+    # Send immediate response
+    initial_response = {
+        "success": True,
+        "data": {
+            "status": "processing",
+            "message": "Response creation started",
+            "success": True,
+        },
+        "error": None,
+        "metadata": None,
+    }
+
+    # Schedule background task
+    background_tasks.add_task(
+        process_response, request, client, assistant, _current_user.organization_id
+    )
+    logger.info(
+        f"[responses.responses] Background task scheduled for response processing: assistant_id={request.assistant_id}, project_id={request.project_id}, organization_id={_current_user.organization_id}"
+    )
+
+    return initial_response
+
+
 @router.post("/responses/sync", response_model=ResponsesAPIResponse)
```

Comment:
Python `Exception` doesn't take key word arguments. Can we update the exception in this try catch ? Also we dont return exception, we raise exceptions

```
    return Exception(error=handle_openai_error(e))
TypeError: Exception() takes no keyword arguments
```